Framework/DataObjects/src/LeanElasticPeak.cpp

```diff
@@ -93,7 +93,9 @@ LeanElasticPeak::LeanElasticPeak(const Geometry::IPeak &ipeak)
 void LeanElasticPeak::setWavelength(double wavelength) { m_wavelength = wavelength; }
 
 /** Return a shared ptr to the reference frame for this peak. */
-std::shared_ptr<const Geometry::ReferenceFrame> LeanElasticPeak::getReferenceFrame() const { return m_refFrame; }
+std::shared_ptr<const Geometry::ReferenceFrame> LeanElasticPeak::getReferenceFrame() const {
+  return (m_refFrame) ? m_refFrame : std::make_shared<const ReferenceFrame>();
```

Comment:
Since this method is doing more than just what its name would suggest, could you add in the documentation that the method will create a default reference frame if the peak doesn't already have one, then return it?